cmd/fluent-watcher/fluentbit/main.go

```diff
@@ -268,10 +275,26 @@ func stop() {
 	}
 
 	if err := cmd.Process.Signal(syscall.SIGTERM); err != nil {
-		_ = level.Info(logger).Log("msg", "Kill Fluent Bit error", "error", err)
+		_ = level.Info(logger).Log("msg", "Error while terminating FluentBit", "error", err)
 	} else {
-		_ = level.Info(logger).Log("msg", "Killed Fluent Bit")
+		_ = level.Info(logger).Log("msg", "Sent SIGTERM to FluentBit, waiting max "+flbTerminationTimeout.String())
 	}
+
+	// Wait for FluentBit to exit, send SIGKILL if it doesn't within the specified timeframe
+	terminated := make(chan *os.ProcessState)
+	go func() {
+		procState, _ := cmd.Process.Wait()
```

Comment:
When the program runs to this, the `fluent-bit` already received the `SIGTERM` signal, if the `fluent-bit` already exit before then, then the `cmd.Wait()` will return, and the `cmd.Process` will be set to nil. In this case call to `cmd.Process.Wait()` will have a `SIGSEGV` error. Maybe it needs to determine the `cmd.Process` is not nil?

Reply:
I haven't noticed this happening (I think it always gets there before fluent-bit actually terminates) but you're correct, this could happen. I will have a look at it. I will also delete "cmd = nil" below that as it's redundant.